### PR TITLE
Use common translation package

### LINKED ISSUE #2273: Switch models using astrotomic/laravel-translatable to spatie/laravel-translatable
Translation tables need to be added to the original as JSON. (i.e. Criteria_Translations 'description' and 'specificity' columns are added to Criteria as JSON, including both locales and translation strings). 

- [x] Edit models
- [x] Database migrations
- [x] Fix factories, unit tests, etc

### PR DESCRIPTION
**Pull request by gggrant**
Resolves #2273.

### CODE REVIEW

**Comment by gggrant** [comment]
Review for Laravel is more/less done. Still fixing bugs in React.

**Comment by Khristophor** [comment]
Ooookay @tristan-orourke @yonikid15 I think this is now good, although I'm not 100% sure how the JPB functioned prior to this regarding locale entries for various things.

I'd also like folks to test out applications, profiles, and the admin portal to see if we have any issues around translated models. Highest risk is probably Skills at this point.

**Comment by yonikid15** [comment]
> Ooookay @tristan-orourke @yonikid15 I think this is now good, although I'm not 100% sure how the JPB functioned prior to this regarding locale entries for various things.
> 
> I'd also like folks to test out applications, profiles, and the admin portal to see if we have any issues around translated models. Highest risk is probably Skills at this point.

Just saw this. Will take a look now 👍

File: app/Http/Controllers/ApplicationByJobController.php [hunk 1]
```diff
@@ -18,8 +18,7 @@
 use App\Models\SkillDeclaration;
 use App\Models\WorkExperience;
 use App\Services\Validation\ApplicationValidator;
-use Facades\App\Services\WhichPortal;
-use Illuminate\Http\Request;
```
Comment (yonikid15) [inline app/Http/Controllers/ApplicationByJobController.php]: Submitting an application broke. Looks like `use Illuminate\Http\Request` is used in `submit()` and `updateAssetSkills()` method.

File: resources/assets/js/components/JobBuilder/WorkEnv/WorkEnvForm.tsx [hunk 2]
```diff
@@ -666,11 +666,17 @@ const updateJobWithValues = (
...
+      ...job.work_env_description,
+      [locale]: envDescription || null,
+    },
+    culture_summary: {
+      ...job.work_env_description,
+      [locale]: cultureSummary || null,
+    },
+    culture_special: {
+      ...job.work_env_description,
+      [locale]: moreCultureSummary || null,
```
Comment (yonikid15) [inline resources/assets/js/components/JobBuilder/WorkEnv/WorkEnvForm.tsx]: The wrong value is getting spread into `culture summary` and `culture_special`.

File: resources/assets/js/components/JobBuilder/Tasks/JobTasks.tsx [hunk 3]
```diff
@@ -129,29 +129,28 @@ const JobTasks: React.FunctionComponent<JobTasksProps &
           if (keyTask) {
             return {
               ...keyTask,
-              [locale]: {
-                description: task.description,
+              description: {
+                en: locale === "en" ? task.description : "",
+                fr: locale === "fr" ? task.description : "",
```
Comment (yonikid15) [inline resources/assets/js/components/JobBuilder/Tasks/JobTasks.tsx]: This is setting the other locale value to null when saved. Try changing it to: `locale === "en" ? task.description : keyTask.description.en` (and same for french).

**Pull request merged**